### PR TITLE
Generator type API

### PR DESCRIPTION
**Pull request by Dinolek**

### CODE REVIEW

File: fabric-generator-type-api-v1/src/main/java/net/fabricmc/fabric/api/generatortype/v1/FabricBiomeSource.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+package net.fabricmc.fabric.api.generatortype.v1;
+
+import java.util.List;
+
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.source.BiomeSource;
+
+public abstract class FabricBiomeSource extends BiomeSource {
```
Comment (i509VCB) [inline fabric-generator-type-api-v1/src/main/java/net/fabricmc/fabric/api/generatortype/v1/FabricBiomeSource.java:25]: This is not meant to be in api?

File: fabric-generator-type-api-v1/src/main/java/net/fabricmc/fabric/api/generatortype/v1/FabricChunkGenerator.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+package net.fabricmc.fabric.api.generatortype.v1;
+
+import java.util.function.Supplier;
+
+import net.minecraft.world.biome.source.BiomeSource;
+import net.minecraft.world.gen.chunk.ChunkGenerator;
+import net.minecraft.world.gen.chunk.ChunkGeneratorSettings;
+
+public abstract class FabricChunkGenerator<T extends FabricGeneratorConfig> extends ChunkGenerator {
```
Comment (i509VCB) [inline fabric-generator-type-api-v1/src/main/java/net/fabricmc/fabric/api/generatortype/v1/FabricChunkGenerator.java:25]: Also this is not meant for api?

File: fabric-generator-type-api-v1/src/main/java/net/fabricmc/fabric/api/generatortype/v1/FabricCustomizationScreen.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.world.CreateWorldScreen;
+import net.minecraft.text.Text;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+@Environment(EnvType.CLIENT)
+public abstract class FabricCustomizationScreen<T extends FabricGeneratorConfig> extends Screen {
```
Comment (i509VCB) [inline fabric-generator-type-api-v1/src/main/java/net/fabricmc/fabric/api/generatortype/v1/FabricCustomizationScreen.java:29]: The name of this screen is arguably too vague. Maybe name it `GenerationCustomizationScreen`

Comment (i509VCB) [inline fabric-generator-type-api-v1/src/main/java/net/fabricmc/fabric/api/generatortype/v1/FabricCustomizationScreen.java:29]: Is there a reason you need to require use of a specific config class here? I believe there is a way to do this without requiring that generic.

**Review by i509VCB — COMMENTED** [review]
I will need to do a further review in the near future. Just a few things I immediately noticed

**Comment by Technici4n** [comment]
Closing due to this PR being seemingly abandoned. If you are still interested in contributing and want this PR reopened, please add a description, and comment so I get notified.

**Pull request closed without merging**